nemo_rl/environments/genrm_environment.py

```diff
@@ -0,0 +1,277 @@
+            if len(individual_scores) != 2:
+                return False, None, None, f"Format violation: Expected exactly 2 individual scores, found {len(individual_scores)}"
+            
+            # Extract preference ranking
+            preference_content = ranking_match.group(1).strip()
+            # Preference ranking should be a single value
+            if ',' in preference_content:
+                return False, None, None, "Format violation: Preference ranking should be a single value, found comma"
+            
+            return True, individual_scores, preference_content, ""
+        
+        else:
+            return False, None, None, f"Unsupported number of responses: {num_responses}"
+    
+    def step(self, message_log_batch: list[list[dict[str, str]]], metadata: list[GenRMEnvironmentMetadata],) -> EnvironmentReturn:
+        """Evaluate GenRM predictions and return rewards."""
+        
+        rewards = []
+        observations = []
+        
+        for conversation, meta in zip(message_log_batch, metadata):
+            # Extract assistant's response
+            assistant_response = ""
+            for msg in conversation:
+                if msg["role"] == "assistant":
+                    assistant_response = msg["content"]
+                    break
+            
+            
+            
+            is_valid_format, individual_scores, preference_ranking, error_message = self.validate_format_and_extract(
+                assistant_response, meta["num_responses"]
+            )
+            # Validate format and extract scores
+            print("assistant_response: ", assistant_response)
+            print("extracted results: ")
+            if individual_scores is not None:
+                print(individual_scores)
+            if preference_ranking is not None:
+                print(preference_ranking)
+            print()
+            
+            
+            if not is_valid_format:
+                # Apply format penalty
+                reward = self.format_penalty
+                rewards.append(float(reward))
+                observations.append({
+                    "role": "environment",
+                    "content": f"Format violation penalty applied. Error: {error_message}. Reward: {reward}"
+                })
+                
+            else:
+                reward = self.format_penalty
+                try:
+                    if meta["num_responses"] == 1:
+                        if meta["helpfulness_1"] is not None and individual_scores:
+                            reward = -self.distance_abs(individual_scores[0], meta["helpfulness_1"])
+                    
+                    elif meta["num_responses"] == 2:
+                        # Calculate distance for both individual scores
+                        if meta["helpfulness_1"] is not None and individual_scores and len(individual_scores) == 2 and meta["preference_ranking"] is not None and preference_ranking:
```

Comment:
Can we make sure that `meta["helpfulness_*"]` is never `None`? We can't learn from those samples anyway so having some would just "pollute" the dataset.